castervoice/rules/core/engine_manager_rules/mic_rules.py

```diff
@@ -0,0 +1,24 @@
+from dragonfly import Dictation, MappingRule, Choice, Function
+from castervoice.lib.actions import Text
+
+from castervoice.lib.ctrl.mgr.rule_details import RuleDetails
+from castervoice.lib.merge.state.short import R
+
+from castervoice.lib.ctrl.mgr.engine_manager import EngineModesManager
+
+class CasterMicRule(MappingRule):
+    mapping = {
+        "caster <mic_mode>": Function(lambda mic_mode: EngineModesManager.set_mic_mode(mode=mic_mode)),     
+    }
+    extras = [
+        Choice(
+            "mic_mode", {
+                "off": "off",
```

Comment:
I get this when saying "caster off"

`Caster: 'off' is not valid. set_mic_state modes are: 'off' - DPI Only, 'on', 'sleeping'`

Is that deliberate? We should also have "caster wake" as an option.

Reply:
@LexiconCode thanks for doing this!

Reply:
Yes it is intentional ('off' - DPI Only). For other engines besides DPI/DNS to have an off state they need a method to turn the microphone on without voice activation like a keyboard shortcut/GUI toggle. DPI/DNS has that capability.